app/viewmodels/pagination/PaginationViewModel.scala

```diff
@@ -0,0 +1,130 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.pagination
+
+import config.AppConfig
+import play.api.i18n.Messages
+import uk.gov.hmrc.govukfrontend.views.viewmodels.pagination.{Pagination, PaginationItem, PaginationLink}
+
+trait PaginationViewModel {
+
+  val results: MetaData
+  val previous: Option[PaginationLink]
+  val next: Option[PaginationLink]
+  val items: Seq[PaginationItem]
+  val pageNumber: Int
+
+  def searchResult(searchParam: Option[String] = None)(implicit messages: Messages): String =
+    (searchParam, results.count) match {
+      case (Some(value), 1) => messages("pagination.number-of-movements.singular.with-search-param", "<b>1</b>", value)
+      case (Some(value), x) => messages("pagination.number-of-movements.plural.with-search-param", s"<b>$x</b>", value)
+      case (None, 1) => messages("pagination.number-of-movements.singular", "<b>1</b>")
+      case (None, x) => messages("pagination.number-of-movements.plural", s"<b>$x</b>")
+    }
+
+  def paginatedSearchResult(searchParam: Option[String] = None)(implicit messages: Messages): String =
+    searchParam match {
+      case Some(value) =>
+        messages("pagination.results.search",
+          s"<b>${results.from}</b>",
+          s"<b>${results.to}</b>",
+          s"<b>${results.count}</b>",
+          value)
+
+      case None =>
+        messages("pagination.results",
+          s"<b>${results.from}</b>",
+          s"<b>${results.to}</b>",
+          s"<b>${results.count}</b>")
+    }
+
+  val pagination: Pagination = Pagination(Some(items), previous, next)
+}
+
+object PaginationViewModel {
+
+  def apply[T <: PaginationViewModel](totalNumberOfItems: Int,
+                                      currentPage: Int,
+                                      numberOfItemsPerPage: Int,
+                                      href: String,
+                                      additionalParams: Seq[(String, String)])
+                                     (f: (MetaData, Option[PaginationLink], Option[PaginationLink], Seq[PaginationItem]) => T)
+                                     (implicit config: AppConfig): T = {
+
+    val results: MetaData = MetaData(totalNumberOfItems, numberOfItemsPerPage, currentPage)
+    val isNumberOfItemsInsideLimit = totalNumberOfItems <= config.numberOfRecordsToDisableNavigationButtonsInPagination
+
+    val hrefWithParams: Int => String = page => additionalParams.foldLeft(s"$href?page=$page") {
```

Comment:
It's a **val** and its scope is just within apply method. Furthermore, it uses additionalParams. If this function have to be moved outside of apply, then it has knock-on effect on other methods and vals in the apply method. Also it has only two statements and easily readable. Let me know in case of any further queries.